backends/ebpf/bpfinclude/ebpf_registry.c

```diff
@@ -0,0 +1,77 @@
+#include "ebpf_user.h"
+
+#define VAR_SIZE 32
+
+struct map_register {
+    char name[VAR_SIZE];        // name of the map
+    struct bpf_map_def *map;    // ptr to the map
+    UT_hash_handle hh;          // makes this structure hashable
+};
+
+struct map_register *maps = NULL;
+
+struct bpf_map_def *registry_lookup(const char *name) {
+    struct map_register *tmp_reg;
+    HASH_FIND(hh, maps, name, VAR_SIZE, tmp_reg);
+    if (tmp_reg==NULL)
+        return NULL;
+    return tmp_reg->map;
+}
+
+int registry_add(struct bpf_map_def *map) {
+    struct map_register *tmp_reg = (struct map_register*)malloc(sizeof(struct map_register));
+    HASH_ADD_KEYPTR(hh, maps, map->name, VAR_SIZE, tmp_reg);
+    tmp_reg->map = map;
+    return 0;
+}
+
+void *bpf_map_lookup_elem(struct bpf_map_def *map, void *key) {
+    struct bpf_map *tmp_map;
+    HASH_FIND(hh, map->bpf_map, key, map->key_size, tmp_map);
+    if (tmp_map==NULL)
+        return NULL;
+    return tmp_map->value;
+}
+
+int bpf_map_update_elem(struct bpf_map_def *map, void *key, void *value,
+                  unsigned long long flags) {
```

Comment:
what are the flags for?

Reply:
The flags define different behaviour regarding updates in the hash map.
#define BPF_ANY       0 /* create new element or update existing */
#define BPF_NOEXIST   1 /* create new element only if it didn't exist */
#define BPF_EXIST     2 /* only update existing element */

I initially just added the exact same prototype as the kernel version which is why the flags argument is still there. We can either add behaviour which emulates the kernel version (https://elixir.bootlin.com/linux/v4.17-rc7/source/kernel/bpf/hashtab.c#L796) or just scrap the flags entirely since we only support one type (BPF_ANY).